nextcord/guild.py

```diff
@@ -2621,14 +2585,14 @@ async def create_role(
             fields['name'] = name
 
         if icon is not MISSING:
-            if icon is None:
-                fields['icon'] = icon
-            elif isinstance(icon, str):
+            if isinstance(icon, str):
```

Comment:
In Discord api docs I saw this.
![image](https://user-images.githubusercontent.com/39167433/164034899-8d229e67-fef4-4941-b362-6a8b186132dd.png)

By default, the icon is `null`. So we don't have to pass `None`. If we want no icon for our new role then we don't need to feed the icon field in the create_role() method, easy. It'll automatically check with `MISSING` and ignore the field.

But if still a user passes create_role(icon=None) it'll give a warning.
![pycharm warning](https://user-images.githubusercontent.com/39167433/164036029-c998c4db-5f63-4eb2-aad0-f663eb45da71.png)

And eventually, at runtime it'll fail because it'll hit the `await icon.read()` which is actually `await None.read()`
![Error if passes None](https://user-images.githubusercontent.com/39167433/164034671-6866a54a-eb3f-48aa-b400-3d8ffd835972.png)

Do you think we should do this? which doesn't make sense to me though.
```py
if icon is not MISSING:
    if icon is None:
        pass
    elif isinstance(icon, str):
        fields['unicode_emoji'] = icon
    elif isinstance(icon, bytes):
        fields['icon'] = utils._bytes_to_base64_data(icon)
    elif isinstance(icon, File):
        fields['icon'] = utils._bytes_to_base64_data(icon.fp.read())
    else:
        fields['icon'] = utils._bytes_to_base64_data(await icon.read())
```

Or this looks better to me without using the `MISSING`👇

```py
async def create_role(
    self,
    *,
    name: str = MISSING,
    permissions: Permissions = MISSING,
    color: Union[Colour, int] = MISSING,
    colour: Union[Colour, int] = MISSING,
    hoist: bool = MISSING,
    mentionable: bool = MISSING,
    icon: Union[str, bytes, File, Asset, Attachment] = None,
    reason: Optional[str] = None
) -> Role:
....

        if icon is not None:
            if isinstance(icon, str):
                fields['unicode_emoji'] = icon
            elif isinstance(icon, bytes):
                fields['icon'] = utils._bytes_to_base64_data(icon)
            elif isinstance(icon, File):
                fields['icon'] = utils._bytes_to_base64_data(icon.fp.read())
            else:
                fields['icon'] = utils._bytes_to_base64_data(await icon.read())
```

Reply:
I think it should actually still set it `None` if `None` is passed, since Discord accepts null (and they _could_ technically change their default).

It fits the API specification better if allows passing None explicitly.

Example:

```py
if icon is None or isinstance(icon, str)
    fields['unicode_emoji'] = icon
```